macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,277 @@
+meta:
+  id: mac_os_resource_snd
+  title: Classic MacOS Sound Resource
+  application: Sound Manager
+  endian: be
+  license: MIT
+  xref:
+    wikidata: Q7564684
+doc-ref: "https://developer.apple.com/library/archive/documentation/mac/pdf/Sound/Sound_Manager.pdf"
+seq:
+  - id: format
+    type: u2
+  - id: num_data_formats
+    -orig-id: number_of_data_formats
+    type: u2
+    if: format == 1
+  - id: data_formats
+    type: data_format
+    repeat: expr
+    repeat-expr: num_data_formats
+    if: format == 1
+  - id: reference_count
+    type: u2
+    if: format == 2
+  - id: num_sound_commands
+    -orig-id: number_of_sound_commands
+    type: u2
+  - id: sound_commands
+    type: sound_command
+    repeat: expr
+    repeat-expr: num_sound_commands
+types:
+  data_format:
+    seq:
+      - id: id
+        type: u2
+        enum: data_type
+      - id: options
+        type: u4
+  sound_command:
+    seq:
+      - id: raw_cmd
+        type: u2
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      cmd:
+        value: raw_cmd & 0x7FFF
+        enum: cmd_type
+      is_data_offset:
+        value: raw_cmd & 0x8000
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset > 0 and cmd == cmd_type::buffer_cmd
```

Comment:
I recommend using [bit-sized integers](https://doc.kaitai.io/user_guide.html#_bit_sized_integers) here, instead of reading a `u2` integer `raw_cmd` and then extracting `cmd` and `is_data_offset` using bitwise AND operators:

```suggestion
    seq:
      - id: is_data_offset
        type: b1
      - id: cmd
        type: b15
        enum: cmd_type
      - id: raw_cmd
        type: u2
      - id: param1
        type: u2
      - id: param2
        type: u4
    instances:
      sound_header_type:
        pos: param2 + 20
        type: u1
        enum: sound_header_type
        if: is_data_offset and cmd == cmd_type::buffer_cmd
      sound_header:
        pos: param2
        size-eos: true
        if: is_data_offset and cmd == cmd_type::buffer_cmd
```

Note that the `b1` type gets implicitly converted to boolean type (`true`/`false`), so you can omit the `is_data_offset > 0` check and use `is_data_offset` as a normal boolean. See the [User Guide > Basic data types](https://doc.kaitai.io/user_guide.html#_basic_data_types):

> **Booleans** can be specified as literal `true` and `false` values as in most languages, but also can be derived by using `type: b1`. This method parses a single bit from a stream and represents it as a boolean value: 0 becomes false, 1 becomes true. This is very useful to parse flag bitfields, as you can omit `flag_foo != 0` syntax and just use something more concise, such as `is_foo`.